packages/toolpad-app/src/toolpad/AppEditor/NodeNameEditor.tsx

```diff
@@ -25,13 +26,19 @@ export default function NodeNameEditor({ node, sx }: NodeNameEditorProps) {
   const nodeNameError = useNodeNameValidation(nameInput, existingNames, node.type);
   const isNameValid = !nodeNameError;
 
-  const handleNameCommit = React.useCallback(() => {
+  const handleNameCommit = React.useCallback(async () => {
+    const oldname = dom.nodes[node.id];
     if (isNameValid) {
-      domApi.setNodeName(node.id, nameInput);
+      if (nameInput !== oldname.name) {
+        domApi.setNodeName(node.id, nameInput);
+      }
     } else {
       setNameInput(node.name);
     }
-  }, [isNameValid, domApi, node.id, node.name, nameInput]);
+    if (nameInput !== oldname.name) {
+      await client.mutation.deletePage(oldname.name);
```

Comment:
`NodeNameEditor` is used in several other places than pages. e.g. https://github.com/mui/mui-toolpad/blob/564108a247a30b6a570974308732020a6e1d479b/packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentEditor.tsx#L175

Wouldn't this break other usage of this component? I can't imagine a solution where this component would need to import 
```tsx
import client from '../../api';
```